### ISSUE TITLE
Svg Beschriftungen 

### ISSUE THREAD

**Issue by Konradrundfunk** [DARC-e-V/50ohm-pocket#10]
Die Svg Beschriftungen sind nur schlecht lesbar. @myzinsky 
![grafik](https://github.com/Konradrundfunk/50ohm-pocket/assets/33392939/9c3248ca-21b5-4b2c-85b3-bb749d14e358)

**Comment by einergehtnochrein** [comment]
Zwar schon lange offen, aber ich kommentiere trotzdem noch was: Mit meinen schlechten Augen auf einem Bildschirm mit hoher Pixeldichte war für mich mit dem Lesen Schluß. Ich habe dann tatsächlich eine Lupe aus der Schublade holen müssen...

Meiner Ansicht nach krankt es neben den Texten auch an weiteren Grafikelementen (der Pfeil im Bild oben ist für mein Sehvermögen ungeeignet). Auch manch andere Proportionen sind nicht wirklich ideal.

Wie erzeugt ihr die SVGs? Werden die noch anderswo eingesetzt, d.h. welche Randbedingungen gibt es wenn man etwas ändern möchte?

**Comment by myzinsky** [comment]
Die SVGs werden automatisch aus Latex Tikz erzeugt. Der Code existiert schon um verschiedene Ansichten für verschiedene Endgeräte zu erzeugen. Es muss generiert und eingebaut werden.

**Comment by einergehtnochrein** [comment]
D.h. dieser Konvertierer oder ggf. auch die Tikz-Quellen wären die Schrauben an denen man drehen müßte...

**Comment by myzinsky** [comment]
Exakt das konnte ich aber machen wenn du mir sagst welche Display breiten du brauchst

**Comment by einergehtnochrein** [comment]
So freihändig könnte ich jetzt nicht sagen was und wo anzupassen wäre.

Nimm mal das Bild am Anfang dieses Threads. Der Strompfeil ist mit Sicherheit gewaltig zu klein in Relation zur übrigen Zeichnung, da wäre dann eine Anpassung in Tikz nötig.

Ob man die Skalierung der SVG-Ausgabe an der Bildschirmgröße festmachen sollte weiß ich nicht. Ist schließlich ein Vektorformat das sich anpassen läßt. Mir scheint allerdings daß die Grafiken auf unterschiedlichen Bildschirmen nicht immer in gleichem Verhältnis zu anderen UI-Elementen angezeigt werden, das wäre dann eher eine Aufgabe für die App.

Ich habe gesehen daß in den SVGs keine `<text>` Tags verwendet werden, sondern der Text wohl als Grafikvektor umgesetzt ist. Wäre das kein Ansatz um die Texte im Bild besser skalierbar zu machen? (Ich bin sicher das macht das Erstellen der Grafiken nicht leichter...)

**Comment by myzinsky** [comment]
Genau das macht unser Algorithmus in LaTeX. Der Sonderfall hier ist, dass das Bild fur eine 4x4 Matrix im Fragenkatalog gerendert wurde (2-3cm Breite):

<img width="422" alt="image" src="https://github.com/user-attachments/assets/3a22a5b8-1700-4e0c-9292-fd8a6f7ef638">

Wenn ich es auf die Bildschirmbreite eines Smartphones rendere wird es natürlich entsprechend größer. Die Schriftgröße bleibt dabei aber immer gleich groß, was ein Designvorteil ist:

<img width="281" alt="image" src="https://github.com/user-attachments/assets/284ea801-fc76-4faf-9a31-498cef4fa745">

Man müsste alle Bilder dann entsprechend nochmal für die Bildischrmbreiten rendern, welche Erforderlich sind. iPhone, iPad, VR-Brillen ... you name it ;-)

**Comment by myzinsky** [comment]
> Als Vektorformat sollte man ja eigentlich mit einem einzigen SVG auskommen.

Nein, dann skaliert man die Symbole. Das wollen wir nicht. Die Symbole und die Schrift sollen immer gleich groß sein. Wenn man das SVG einfach nur skaliert skalieren die Schriften und Symbole immer mit. Mit unserem Ansatz können wir den Abstand zwischen den Komponenten kann man immer skalieren, wie bei einem Gummiband. Ich hoffe mit der Analogie wird es klarer? :-) Oder hat SVG auch so ein Feature?

Ansonsten erklären Marei und ich das auch hier: https://youtu.be/RywhJMeg2LE?si=bpZEEblUZzcpLx4C&t=1059

**Comment by einergehtnochrein** [comment]
Wow, die Komplexität wird ein klein wenig spürbar... Beeindruckend welche Arbeit ihr da rein steckt!

Also unser Lösungsweg wird dann sein, beizeiten spezialisierte SVGs für die Trainingsapp aus dem Directus-Nexus zu erhalten?

**Comment by myzinsky** [comment]
Genau, aber nicht nur ein SVG pro bild. Sondern mehrere für Tablets, Telefone und was noch so kommt :-)

**Comment by fritzsche** [comment]
DIe unlesbaren Bilder habe ich auch festgestellt. Ich kann die Erklärung von Marei nachvollziehen, aber im Moment ist die Darstellung so schlecht (klein), dass die App deutlich profitieren würde, wenn die Bilder einfach nur auf den Verfügbaren Platz skalieren.  
Auch Im Querformat wird es nicht besser (aber riesige leere Ränder):
 
![Simulator Screen Shot - iPhone SE (3rd generation) - 2024-10-12 at 00 52 08](https://github.com/user-attachments/assets/2ce3658a-eeaa-47f9-8523-d4fcdff8a546)

(hier auch ein Fehler mit X_3). 
Die 50 Ohm App sollte auch touch unterstützen: Zoom via pinch immerhin ein großer Vorteil von Smartphones.
Klar müsste man große Grafiken (z.B: Schaltpläne in der Technik A, z.B. AD408). ggf. speziell für verschiedenen Mobil-Plattformen der App neu Zeichnen, aber prio vielleicht doch erstmal überhaupt lesbar zu sein?

**Comment by Konradrundfunk** [comment]
Ja das sehe ich auch so, ich hatte schon mit pinch to zoom direkt in der Ansicht eperimentiert aber das ist sehr herausfordernd. Deshalb denke ich es ist am besten den Nutzenden auf eine neue Seite weiterzuleiten in der man zoomen kann. Was denkt ihr dazu_

**Comment by myzinsky** [comment]
Skalier doch einfach die Bilder erstmal auf die volle Breite. Dann sollte es schon viel helfen.

**Comment by Konradrundfunk** [comment]
Ja koente man machen dann wird aber der vertikale Platz sehr stark eingenommen, vorallem wenn das Geraet horizontal genutzt wird.

**Comment by fritzsche** [comment]
nur Info: In meinem kleinen Projekt hatte ich überraschende Schwierigkeiten
mit der richtigen Größe der Bilder. Einfaches skalieren (z.B. 100% Breite) hat nicht funktioniert, da z.B. die Steckverbinder (BNC/N/UHF etc.) viel zu groß wurden. Ich habe einige Stunden verschiedene Kombinationen probiert und ein großzügiger konstanter Skalierungsfaktor bei gleichzeitiger dynamischer maximaler Größe sah noch am besten aus. Hier die CSS Definition;

https://github.com/fritzsche/afu_test/blob/1022cd0dd057e8e074b2e9e6e6505027248e5f0a/style.css#L14C1-L20C2

**Comment by myzinsky** [comment]
Für den Fragenkatalog und für das Buch haben wir in unserer Datenbank einen Scaling-Hint, den wir hier nutzen. Wir könnten überlegen, diese Hints als ein zusätzliches JSON bereit zu stellen.

**Comment by fritzsche** [comment]
https://github.com/DARC-e-V/50ohm-pocket/pull/54
